keyboards/planck/keymaps/buhearns/keymap.c

```diff
@@ -0,0 +1,189 @@
+    KC_TRNS,     EEP_RST,  KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   MUV_DE,  MU_OFF,  AU_OFF,   KC_TRNS,
+    KC_TRNS,     KC_NO,    KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_MUTE, KC_MPRV, KC_MNXT,  KC_NO,
+    KC_TRNS,     KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MSTP, KC_VOLD, KC_VOLU,  KC_MPLY
+),
+
+[5] = LAYOUT_planck_grid(
+    B_DE_C,     B_DE_1,    B_DE_2,  B_DE_3,  B_DE_S,  B_DE_E,  KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,    KC_NO,
+    KC_TRNS,    B_DEAB,    B_DEOB,  B_DEUB,  B_DEQR,  B_DEQL,  KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,    KC_NO,
+    KC_TRNS,    B_DEAS,    B_DEOS,  B_DEUS,  B_DE_B,  KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,    KC_NO,
+    KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS
+)};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case B_UPD:
+      if (record->event.pressed) {
+        SEND_STRING("sudo apt update && sudo apt full-upgrade -yyy" SS_TAP(X_ENTER));
+      }
+      break;
+    case B_SSH:
+      if (record->event.pressed) {
+        SEND_STRING("cd /etc/ssh" SS_TAP(X_ENTER) "sudo shasum ssh_host_*" SS_TAP(X_ENTER) "sudo rm ssh_host_*" SS_TAP(X_ENTER) "sudo dpkg-reconfigure openssh-server" SS_TAP(X_ENTER));
+      }
+      break;
+    case B_SSHK:
+      if (record->event.pressed) {
+        SEND_STRING("ssh-keygen -t ed25519" SS_TAP(X_ENTER) SS_TAP(X_ENTER) SS_TAP(X_ENTER) SS_TAP(X_ENTER));
+      }
+      break;
+    case B_USR:
+      if (record->event.pressed) {
+        SEND_STRING("useradd -m user" SS_TAP(X_ENTER) "usermod -a -G sudo user" SS_TAP(X_ENTER) "chsh -s /bin/bash user" SS_TAP(X_ENTER) "passwd user" SS_TAP(X_ENTER));
+      }
+      break;
+    case B_X11:
+      if (record->event.pressed) {
+        SEND_STRING("sudo x11vnc -storepasswd /etc/x11vnc.pass" SS_TAP(X_ENTER));
+      }
+      break;
+    case B_X112:
+      if (record->event.pressed) {
+        SEND_STRING("sudo nano /lib/systemd/system/x11vnc.service" SS_TAP(X_ENTER) "[Unit]" SS_TAP(X_ENTER) "Description=Start X11VNC" SS_TAP(X_ENTER) "After=multi-user.target" SS_TAP(X_ENTER) SS_TAP(X_ENTER) "[Service]" SS_TAP(X_ENTER) "Type=simple" SS_TAP(X_ENTER) "ExecStart=/usr/bin/x11vnc -display :0 -auth guess -forever -loop -noxdamage -repeat -rfbauth /etc/x11vnc.pass -rfbport 5900 -shared -localhost -o /var/log/x11vnc.log" SS_TAP(X_ENTER) SS_TAP(X_ENTER) "[Install]" SS_TAP(X_ENTER) "WantedBy=multi-user.target");
+      }
+      break;
+    case B_X113:
+      if (record->event.pressed) {
+        SEND_STRING("sudo systemctl enable x11vnc.service" SS_TAP(X_ENTER));
+      }
+      break;
+    case B_DEAB:
+      if (record->event.pressed) {
+        SEND_STRING(SS_DOWN(X_RALT) SS_TAP(X_KP_0) SS_TAP(X_KP_1) SS_TAP(X_KP_9) SS_TAP(X_KP_6) SS_UP(X_RALT));
```

Comment:
If you  enable unicode, you could handle this much easier.

https://docs.qmk.fm/#/feature_unicode

Reply:
Thank you! I already tried that but didn't realize i had to switch to the UC_WINC mode (as opposed to the UC_WIN mode) for the characters to work.

Everything working now but the Euro sign (0x20AC) and Bullet (0x2022). Am I missing something?